tests/unit/source_code/test_files.py

```diff
@@ -74,6 +94,28 @@ def test_files_walks_directory():
     assert languages.fixer.call_count > 1
 
 
+def test_file_loader_exists():
+    """File exists"""
+    file_loader = FileLoader()
+    assert file_loader.exists(Path(__file__))
+
+
+def test_file_loader_not_exists():
+    """File does not exists"""
+    file_loader = FileLoader()
+    assert not file_loader.exists(Path("/some/path/that/does/not/exist"))
+
+
+@pytest.mark.parametrize("path", [Path("/some/path/that/does/not/exists"), Path("lib.py")])
+def test_file_loader_load_dependency_calls_dependency_load(mock_path_lookup, path):
+    """The file loader load dependency points to the dependency load"""
+    file_loader = FileLoader()
+    dependency = Dependency(file_loader, path)
+    # TODO: Clarify if this is the right way to use it.
+    # It reads a bit convoluted to pass the file_loader to the dependency first and then the dependency to the file.
+    assert file_loader.load_dependency(mock_path_lookup, dependency) == dependency.load(mock_path_lookup)
```

Comment:
@ericvergnaud : Could you clarify this bi-directionality between the `loader.load_dependency` and `dependency.load`? It boggles my mind, maybe we should have one-directional reference, maybe not